### PR TITLE
Check fields for references

### PR DESCRIPTION
**Pull request by br648**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Refactor to allow for internal table references to be correctly merged.

### CODE REVIEW

File: src/main/java/com/conveyal/datatools/manager/jobs/feedmerge/StopsMergeLineContext.java [hunk 1]
```diff
@@ -33,6 +34,26 @@ public boolean checkFieldsForMergeConflicts(Set<NewGTFSError> idErrors, FieldCon
...
+    public void checkFieldsForReferences(FieldContext fieldContext) {
+        updateParentStationReference(fieldContext);
+    }
+
+    /**
+     * If there is a parent station reference, update to include the scope stop_id.
+     */
+    private void updateParentStationReference(FieldContext fieldContext) {
+        Field field = fieldContext.getField();
+        if (field.name.equals("parent_station")) {
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/jobs/feedmerge/StopsMergeLineContext.java:47]: There is a `FieldContext.nameEquals` shorthand method you can use.

**Review by binh-dam-ibigroup — APPROVED** [review]
Seems to be working with UI. See the one minor comment.

File: src/main/java/com/conveyal/datatools/manager/jobs/feedmerge/StopsMergeLineContext.java [hunk 2]
```diff
@@ -33,6 +34,25 @@ public boolean checkFieldsForMergeConflicts(Set<NewGTFSError> idErrors, FieldCon
...
+    }
+
+    /**
+     * If there is a parent station reference, update to include the scope stop_id.
+     */
+    private void updateParentStationReference(FieldContext fieldContext) {
+        if (fieldContext.nameEquals("parent_station")) {
+            String parentStation = fieldContext.getValue();
+            if (!"".equals(parentStation)) {
+                LOG.debug("Updating parent station to: " + getIdWithScope(parentStation));
```
Comment (leonardehrenfried) [inline src/main/java/com/conveyal/datatools/manager/jobs/feedmerge/StopsMergeLineContext.java:49]:
```suggestion
                LOG.debug("Updating parent station to: {}", getIdWithScope(parentStation));
```

With a single parameter it doesn't make much of a difference, but I like this style better.

**Review by leonardehrenfried — APPROVED** [review]

**Comment by br648** [comment]
@binh-dam-ibigroup I'm not authorized to merge this pull request! Can you do the honors please?

**Pull request merged**